### ISSUE TITLE
3.1.0-beta.4: Editor sidebar always loads to the Information tab

### ISSUE THREAD

**Issue by TotallyInformation** [node-red/node-red#4293]
### Current Behavior

When (re)loading the Editor, no matter what sidebar tab was previously loaded, the information sidebar is always loaded.

Tested with Edge, Vivaldi and Firefox. Tested with tab icons in different orders.

Can see Local Storage changing during reload:

Local storage before page reload:
![image](https://github.com/node-red/node-red/assets/1591850/3d78920d-6d2b-415d-be5d-29f9e178a624)

Changes back to:
![image](https://github.com/node-red/node-red/assets/1591850/2d59ae45-8c98-4b53-b190-8698807ebcec)

While the page is loading (with the progress bar still showing).

### Expected Behavior

Previous versions always retained the last-viewed sidebar type.

### Steps To Reproduce

Change order of sidebar tab icons. Select one other than the information sidebar, reload the Editor.

### Example flow

N/A

### Environment

- Node-RED version: 3.1.0-beta.4
- Node.js version: v18.15.0
- npm version: N/A
- Platform/OS: Windows 11
- Browser: Edge, Vivaldi, and Firefox (all latest versions)

**Comment by colinl** [comment]
I am not seeing the issue, for me it always restores the previous tab.
In the browser local storage I see the tab switch to info while loading the page, then go to the previous selected tab.
I am using the same NR version. 
Two node red servers, one Ubuntu 22.04 using nodejs 20.5.1, the other Pi OS with nodejs 18.16.0
Browsers Edge and FF on Ubuntu.

**Comment by Steve-Mcl** [comment]
Same as @colinl, i dont see this in Chrome on Windows.

I wonder if some plugin or node is requesting the debug panel be show?

Perhaps you could do some digging?

Perhaps add some `console.trace();` logging in the source around where the storage item changes?

**Comment by knolleary** [comment]
As with others, I cannot reproduce this locally so we may need to figure out how best to debug what is happening in @TotallyInformation environment.

Here's how it has been implemented:

1. When the editor is initialising, before any sidebar tabs are added, the sidebar container grabs the current value of `last-sidebar-tab` for later - https://github.com/node-red/node-red/blob/2478a7194ede810023bd8a270bcff645a6030426/packages/node_modules/%40node-red/editor-client/src/js/ui/sidebar.js#L269

2. The editor then loads the built-in sidebar tabs, plus all nodes/plugins that contribute sidebar tabs. This can cause `last-sidebar-tab` to be updated in local storage as things settle down. But this isn't an issue as we captured the original value at step 1

3. Once all that has happened, the editor asks the sidebar to show the `:first` tab - this is a keyword used to show whatever was intended to be the first tab - ie what was grabs in step 1. https://github.com/node-red/node-red/blob/2478a7194ede810023bd8a270bcff645a6030426/packages/node_modules/%40node-red/editor-client/src/js/ui/sidebar.js#L207-L209

But note how this code will default to the visibly 'first' tab *if* `lastSessionSelectedTab` is not set.

My theory is that for some reason (maybe security policy?) the attempt to get the value from localStorage is being blocked.

Would be interesting to know what `RED.settings.getLocal("last-sidebar-tab")` returns.
